Functional: Foast parser can't handle empty assigns

See this test: https://github.com/BenWeber42/gt4py/commit/d3cc8d58dc22be2af3e3e9c90135d58dcac24656

```python
def test_empty_assign():
    def empty_assign() -> float:
        x: float
        return 1.0

    _ = FieldOperatorParser.apply_to_function(empty_assign)
```

Fails with:

```console
=================================================================================================================== FAILURES ===================================================================================================================
______________________________________________________________________________________________________________ test_empty_assign _______________________________________________________________________________________________________________

    def test_empty_assign():
        def empty_assign() -> float:
            x: float
            return 1.0
    
>       _ = FieldOperatorParser.apply_to_function(empty_assign)

tests/functional_tests/ffront_tests/test_foast_parsing.py:294: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
src/functional/ffront/dialect_parser.py:115: in apply_to_function
    return cls.apply(src, closure_vars, annotations)
src/functional/ffront/dialect_parser.py:94: in apply
    ).visit(cls._preprocess_definition_ast(definition_ast)),
../ben-stack/epoche-003/spack-instance/opt/spack/linux-rhel7-skylake_avx512/gcc-12.1.0/python-3.10.6-pe6hg5gby4tsmwdb2nwqbosacmpcmp4h/lib/python3.10/ast.py:410: in visit
    return visitor(node)
src/functional/ffront/func_to_foast.py:169: in visit_FunctionDef
    body=self.visit_stmt_list(node.body, **kwargs),
src/functional/ffront/func_to_foast.py:282: in visit_stmt_list
    return [self.visit(node, **kwargs) for node in nodes]
src/functional/ffront/func_to_foast.py:282: in <listcomp>
    return [self.visit(node, **kwargs) for node in nodes]
../ben-stack/epoche-003/spack-instance/opt/spack/linux-rhel7-skylake_avx512/gcc-12.1.0/python-3.10.6-pe6hg5gby4tsmwdb2nwqbosacmpcmp4h/lib/python3.10/ast.py:410: in visit
    return visitor(node)
src/functional/ffront/func_to_foast.py:228: in visit_AnnAssign
    return foast.Assign(
<attrs generated init functional.ffront.field_operator_ast.Assign>:8: in __init__
    __attr_validator_value(self, __attr_value, self.value)
src/eve/datamodels/core.py:202: in __call__
    self.validator(value)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

value = None, kwargs = {}

    def _is_instance_of(value: Any, **kwargs: Any) -> None:
        if not isinstance(value, type_):
>           raise TypeError(
                f"'{name}' must be {type_} (got '{value}' which is a {type(value)})."
            )
E           TypeError: 'Assign.value' must be <class 'functional.ffront.field_operator_ast.Expr'> (got 'None' which is a <class 'NoneType'>).

src/eve/type_validation.py:364: TypeError
=========================================================================================================== short test summary info ============================================================================================================
```

The problem lies here: https://github.com/GridTools/gt4py/blob/c0388b6757a363d51361132b5206e6fac6a5f609/src/functional/ffront/func_to_foast.py#L228-L236
Because `value` can't be `None`: https://github.com/GridTools/gt4py/blob/c0388b6757a363d51361132b5206e6fac6a5f609/src/functional/ffront/field_operator_ast.py#L162-L164

--- a/src/gt4py/next/ffront/func_to_foast.py
+++ b/src/gt4py/next/ffront/func_to_foast.py
@@ -291,6 +291,11 @@ class FieldOperatorParser(DialectParser[foast.FunctionDefinition]):
         if not isinstance(node.target, ast.Name):
             raise errors.DSLError(self.get_location(node), "Can only assign to names.")
 
+        if node.value is None:
+            raise errors.DSLError(
+                self.get_location(node), "Variable declaration without assignment is not allowed."
+            )
+
         if node.annotation is not None:
             assert isinstance(node.annotation, ast.Constant) and isinstance(
                 node.annotation.value, str
@@ -306,7 +311,7 @@ class FieldOperatorParser(DialectParser[foast.FunctionDefinition]):
             target=foast.Symbol[ts.FieldType](
                 id=node.target.id, location=self.get_location(node.target), type=target_type
             ),
-            value=self.visit(node.value) if node.value else None,
+            value=self.visit(node.value),
             location=self.get_location(node),
         )
 

--- a/tests/next_tests/unit_tests/ffront_tests/test_func_to_foast.py
+++ b/tests/next_tests/unit_tests/ffront_tests/test_func_to_foast.py
@@ -132,6 +132,15 @@ def test_invalid_assign_to_expr():
         _ = FieldOperatorParser.apply_to_function(invalid_assign_to_expr)
 
 
+def test_declaration_without_assignment():
+    def empty_assign() -> float:
+        x: float
+        return 1.0
+
+    with pytest.raises(errors.DSLError, match=r"without assignment"):
+        _ = FieldOperatorParser.apply_to_function(empty_assign)
+
+
 def test_temp_assignment():
     def copy_field(inp: gtx.Field[[TDim], "float64"]):
         tmp = inp

